File: java/src/main/java/com/techelevator/services/UserListService.java
package com.techelevator.services;

import com.techelevator.dao.UserListDao;
import com.techelevator.model.UserList;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class UserListService {

    private final UserListDao userListDao;

    public UserListService(UserListDao userListDao) {
        this.userListDao = userListDao;
    }

    public List<UserList> getUserListsByUserId(int userId) {
        return userListDao.getUserListsByUserId(userId);
    }

    public List<UserList> getUserListByGameIdUserId(int gameId, int userId) {
        return userListDao.getUserListByGameIdUserId(gameId, userId);
    }

    public UserList getUserListById(int id) {
        return userListDao.getUserListById(id);
    }

    public UserList createUserList(UserList userList) {
        return userListDao.createUserList(userList);
    }

    public UserList updateUserList(UserList userList) {
        return userListDao.updateUserList(userList);
    }

    public UserList addGameToUserList(int listId, int gameId) {
        return userListDao.addGameToUserList(listId, gameId);
    }

    public UserList removeGameFromUserList(int listId, int gameId) {
        return userListDao.removeGameFromUserList(listId, gameId);
    }

    public boolean deleteUserListById(int id) {
        return userListDao.deleteUserListById(id);
    }

}

File: vue/src/services/UserListService.js
import axios from 'axios';

export default {

  getUserListsByUserId(userId){
    return axios.get(`/user/${userId}/userlists`)
  },

  getUserListByGameIdUserId(userId, gameId) {
    return axios.get(`/user/${userId}/game/${gameId}/userlists`);
  },

  getUserListById(id){
    return axios.get(`userlist/${id}`)
  },

  createUserList(userlist){
    return axios.post(`/userlist/`, userlist)
  },

  editUserList(id, userlist){
    return axios.put(`/userlist/${id}`, userlist)
  },

  addGameToUserList(listId, gameId){
    return axios.post(`userlistAdd/${listId}/${gameId}`)
  },

  removeGameFromUserList(listId, gameId){
    return axios.delete(`userlistRemove/${listId}/${gameId}`)
  },

  deleteUserList(id) {
    return axios.delete(`/userlist/${id}`);
  }
}